libs/wingsdk/src/std/array.ts

```diff
@@ -27,6 +27,28 @@ export class ImmutableArray {
     index;
     throw new Error("Abstract");
   }
+
+  /**
+   * Create a mutable shallow copy of this array
```

Comment:
TBH, why are we doing a shallow copy instead of a deep clone? In most cases you'd want a deep clone...

Maybe we need both?

Reply:
I think we should have both eventually, but the purpose of this PR I think shallow is not only good enough but probably expected. `deep` things are not usually the default due to the cost, so this probably shouldn't be any different.

Reply:
Sounds good to me. Maybe raise an issue and we can gauge need

Reply:
https://github.com/winglang/wing/issues/1450